astropy/table/scripts/showtable.py

```diff
@@ -0,0 +1,111 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+``showtable`` is a command-line script based on ``astropy.io`` and
+``astropy.table`` for printing ASCII, FITS, HDF5 or VOTable files(s) to the
+standard output.
+
+Example usage of ``showtable``:
+
+1. FITS::
+
+    $ showtable astropy/io/fits/tests/data/table.fits
+
+    target V_mag
+    ------- -----
+    NGC1001  11.1
+    NGC1002  12.3
+    NGC1003  15.2
+
+2. ASCII::
+
+    $ showtable astropy/io/ascii/tests/t/simple_csv.csv
+
+     a   b   c
+    --- --- ---
+      1   2   3
+      4   5   6
+
+3. XML::
+
+    $ showtable astropy/io/votable/tests/data/names.xml --max-width 70
+
+               col1             col2     col3  ... col15 col16 col17
+               ---              deg      deg   ...  mag   mag   ---
+    ------------------------- -------- ------- ... ----- ----- -----
+    SSTGLMC G000.0000+00.1611   0.0000  0.1611 ...    --    --    AA
+
+
+
+4. Print all the FITS tables in the current directory::
+
+    $ showtable *.fits
+
+"""
+
+import argparse
+from astropy.table import Table
+from astropy import log
+
+
+def showtable(filename, args):
+    """
+    Read a table and print to the standard output.
+
+    Parameters
+    ----------
+    filename : str
+        The path to a FITS file.
+
+    """
+    read_kwargs = {k: v for k, v in vars(args).items()
+                   if k in ('hdu', 'format', 'table_id') and v is not None}
+    try:
+        table = Table.read(filename, **read_kwargs)
+        formatter = table.more if args.more else table.pprint
+        formatter(max_lines=args.max_lines, max_width=args.max_width,
+                  show_unit=not args.hide_unit, show_dtype=args.show_dtype)
+    except IOError as e:
+        log.error(str(e))
+
+
+def main(args=None):
+    """The main function called by the `showtable` script."""
+    parser = argparse.ArgumentParser(
+        description=('Print tables from ASCII, FITS, HDF5, VOTable file(s).'))
+    addarg = parser.add_argument
+
+    # pprint arguments
+    addarg('--more', action='store_true',
+           help='Use the pager mode from Table.more.')
+    addarg('--max-lines', type=int,
+           help='Maximum number of lines in table output.')
+    addarg('--max-width', type=int,
+           help='Maximum character width of output.')
+    addarg('--hide-unit', action='store_true',
+           help='Hide the header row for unit (which is shown '
+           'only if one or more columns has a unit).')
+    addarg('--show-dtype', action='store_true',
+           help='Include a header row for column dtypes.')
+
+    # ASCII-specific arguments
+    # FIXME: add more args ? (delimiter, guess ?)
```

Comment:
I wonder if it would make sense to generalize this so that all the reader-specific kwargs are not initially pre-defined in the argparse parser.  See e.g. for a hackish way to do this (which BTW doesn't totally work...)
https://stackoverflow.com/questions/37367331/is-it-possible-to-use-argparse-to-capture-an-arbitrary-set-of-optional-arguments

Probably a better option is to pre-code allowed options but use subparsers to keep them tidy and compartmentalized.  E.g.
```
showtable ascii filename.dat --delimiter='|' --quotechar='"' --data-start=5
showtable ascii --help . # shows ascii subparser options
```

Reply:
I'm not sure it's useful to add all possible parameters, my idea was to add only the most frequent ones (`delimiter` is certainly useful). The issue with subparsers is that it prevents a more generic use where you don't have to know the file type in advance, `Table.read` is able to guess many formats. Also having to specify always a subparser is cumbersome !